src/javascript/javascript-import-scanner.ts

```diff
@@ -15,26 +15,95 @@
 import * as babel from 'babel-types';
 import {dirname, relative} from 'path';
 import {URL} from 'whatwg-url';
-import {FileRelativeUrl, ScannedImport, Severity, Warning} from '../model/model';
+
+import {Document, FileRelativeUrl, Import, ResolvedUrl, ScannedImport, Severity, Warning} from '../model/model';
 
 import {Visitor} from './estree-visitor';
 import {JavaScriptDocument} from './javascript-document';
 import {JavaScriptScanner} from './javascript-scanner';
 
 import isWindows = require('is-windows');
 import resolve = require('resolve');
+import {SourceRange} from '../model/model';
 
 const isPathSpecifier = (s: string) => /^\.{0,2}\//.test(s);
 
 export interface JavaScriptImportScannerOptions {
   /**
    * Algorithm to use for resolving module specifiers in import
-   * and export statements when rewriting them to be web-compatible.
+   * and export statements when converting them to URLs.
    * A value of 'node' uses Node.js resolution to find modules.
+   *
+   * If this argument is not given, module specifiers must be web-compatible
+   * urls.
    */
   moduleResolution?: 'node';
 }
 
+export class ScannedJavascriptImport extends ScannedImport {
+  readonly type!: 'js-import';
+
+  // The original text of the specifier. Unlike `this.url`, this may not
+  // be a URL, but may be a bare module specifier, like 'jquery'.
+  readonly specifier: string;
+
+  constructor(
+      url: FileRelativeUrl|undefined, sourceRange: SourceRange|undefined,
+      urlSourceRange: SourceRange|undefined,
+      ast: babel.ImportDeclaration|babel.CallExpression|
+      babel.ExportAllDeclaration|babel.ExportNamedDeclaration,
+      lazy: boolean, originalSpecifier: string) {
+    super('js-import', url, sourceRange, urlSourceRange, ast, lazy);
+    this.specifier = originalSpecifier;
+  }
+
+  protected constructImport(
+      resolvedUrl: ResolvedUrl, relativeUrl: FileRelativeUrl,
+      importedDocument: Document, _containingDocument: Document) {
+    return new JavascriptImport(
+        resolvedUrl,
+        relativeUrl,
+        this.type,
+        importedDocument,
+        this.sourceRange,
+        this.urlSourceRange,
+        this.astNode,
+        this.warnings,
+        this.lazy,
+        this.specifier);
+  }
+}
+
+declare module '../model/queryable' {
+  interface FeatureKindMap {
+    'js-import': JavascriptImport;
+  }
+}
+export class JavascriptImport extends Import {
+  /**
+   * The original text of the specifier. Unlike `this.url`, this may not
+   * be a URL, but may be a bare module specifier, like 'jquery'.
+   */
+  readonly specifier: string;
+  constructor(
+      url: ResolvedUrl, originalUrl: FileRelativeUrl, type: string,
+      document: Document, sourceRange: SourceRange|undefined,
+      urlSourceRange: SourceRange|undefined, ast: any, warnings: Warning[],
```

Comment:
why is ast any?

Reply:
copy-pasta from the definition of `Import`.

We should probably do a sweep of all uses of `any` in the analyzer.